server/methods/core/orders.js

```diff
@@ -850,5 +949,61 @@ Meteor.methods({
       throw new Meteor.Error(
         "Attempt to refund transaction failed", result.error);
     }
+  },
+
+  // important (added to deal with cancelling orders)
+  /**
+   * orders/cancelOrder
+   *
+   * @summary Cancel an order that hasn't been shipped
+   * @param {String} orderId - order object
+   * @return {null} no return value
+   */
+  "orders/cancelOrder": function (orderId) {
+    check(orderId, String);
+
+    if (!Reaction.hasPermission("orders")) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+
+    // order and billing information
+    const order = Orders.findOne(orderId);
+    const paymentMethod = order.billing[0].paymentMethod;
+    const paymentStatus = paymentMethod.status;
+    const amount = paymentMethod.amount;
+    const shipment = order.shipping[0];
+
+    // if payment has been captured
+    if (paymentStatus === "completed") {
+      // Refund money
+      Meteor.call("orders/refunds/create", orderId, paymentMethod, amount, (error) => {
+        if (error) {
+          Alerts.alert(error.reason);
+          return;
```

Comment:
rather than having a `return` statement here to terminate the flow of execution, why not put the `send email to user` block from line 985 into the an else block. This way, you would check for error if any and it won't reach the else block or if no error, it would send the email to the required user. Please reach out if you do not understand my suggestion.